src/containers/App/App.js

```diff
@@ -9,60 +9,72 @@ import Header from '../../containers/Header/Header';
 import MainGallery from '../../components/MainGallery/MainGallery';
 import UserSignup from '../UserMenu/UserSignup';
 import UserLogin from '../UserMenu/UserLogin';
-import Error from '../../components/Error/error404';
+import GenreContainer from '../../components/GenreContainer/GenreContainer'
+// import Error from '../../components/Error/error404';
+import { findGenres, fetchSingleGenre } from '../../utils/API/ApiFetch'
+import Genre from '../../components/Genre/Genre'
 
 export class App extends Component {
-	componentDidMount() {
-		this.getDefaultData();
-	}
+  constructor() {
+    super()
+    this.state={
+      genres:[]
+    }
+  }
+  async componentDidMount() {
+    this.getLoadingMovies();
+    const genres = await findGenres() 
+    this.setState({...genres})
+  }
+  
+  populateRoutes= ()=>  {
+    console.log('populateRoutes')
+    const routes = this.state.genres.map(genre => {
+      return <Route path={`/genre/${genre.name}`} render={(props) => <Genre {...props} genre={genre} />} />
```

Comment:
I like this too. Creating a function to create what would look like dupes is awesome!